source/asynch_socket.c

```diff
@@ -610,9 +610,14 @@ socket_error_t lwipv4_socket_send(struct socket *socket, const void * buf, const
         socket->event = NULL;
         break;
     }
-    case SOCKET_STREAM:
-        err = tcp_write(socket->impl,buf,len,TCP_WRITE_FLAG_COPY);
-        break;
+    case SOCKET_STREAM:{
+            struct tcp_pcb* pcb = socket->impl;
+            err = tcp_write(pcb,buf,len,TCP_WRITE_FLAG_COPY);
+            if (tcp_nagle_disabled(pcb)) {
+                tcp_output(pcb);
```

Comment:
Are you sure this is the right thing to do? lwIP already has a way to disable Nagle:

http://lwip.wikia.com/wiki/Raw/TCP

Calling `tcp_output` after each `tcp_write` might not be equivalent with setting the `TF_NODELAY` flag as far as the stack is concerned.